File: DSA/Queue_implementation.py
class Queue:
    def __init__(self,q):
        self.q = q
    
    def enqueue(self,ele):
        self.q.append(ele)

    def dequeue(self):
        if len(self.q) < 1:
            return False
        return self.q.pop(0)

    def display(self):
        print(self.q)      
q = []
s = Queue(q)
s.display()
s.enqueue(9)
s.display()
s.enqueue(8)
s.display()
s.enqueue(7)
s.display()
s.dequeue()
s.display()
s.dequeue()
s.display()



    

File: DSA/Queue_implementation.java
class Queue{
    int SIZE = 5;
    int front,rear;
    int[] queue;

    public Queue(){
        front = -1;
        rear = -1;
        queue = new int[SIZE];
    }

    public void enqueue(int ele){
        if(rear == SIZE -1){
            System.out.println("queue is full");
        }
        else{
        if(front == -1){
            front = 0;
        }
        rear++;
        queue[rear] = ele;
        System.out.println("element inserted");
    }
    }

    public int dequeue(){
        if(front == -1 || front > rear){
            return -1;
        }
        else{
            front++;
            System.out.println("element removed");

        }
        return queue[front];
    }
    public boolean isEmpty(){
        if(front == -1){
            return true;
        }else{
            return false;
        }
    }


public void display(){
    if(isEmpty()){
        System.out.println("queue is empty");
    }else{
        for(int i = front;i<=rear;i++){
            System.out.println(queue[i]);
        }
    }
}

public static void main(String[] args){
    Queue q = new Queue();
    q.enqueue(7);
    q.display();
    q.enqueue(5);
    q.display();
    q.enqueue(9);
    q.display();
    System.out.println("element deleted"+q.dequeue());

    q.display();


}

}